test/journalpump/test_journalpump.py

```diff
@@ -473,11 +474,18 @@ def test_pump_reader(journald_server: StandaloneJournalD):
     for line in src_logs:
         journald_server.send_log(line)
 
-    entries = []
+    entries: List[JournalObject] = []
+    start_time = time()
     while True:
         entry = pump_reader.get_next()
         if entry is None:
-            break
+            if {entry.entry["MESSAGE"] for entry in entries}.issuperset(set(src_logs)):
+                break
```

Comment:
The lines we collect are always more then what we wrote because the starting up journald process logs diagnostic data (name and memory) at the start. We can't really speak to whether that will change in the future nor is it reasonable to filter it out via this interface. For the purposes of the test the only thing we want to establish is that lines that go in are seen coming out.

Hence, the collection of unique log entries should always be a superset of what we actually sent (and at least equal).